pkg/fetcher/fetcher_test.go

```diff
@@ -0,0 +1,110 @@
+		}]
+	}
+	`
+	MultipleErrJsonStr = `{
+	"errors":
+		[{"code":"UNAUTHORIZED",
+		  "message":"authentication required",
+		  "detail":[{"Type":"repository","Class":"","Name":"library/jiowengew","Action":"pull"}]
+		},
+		{"code":"NOTFOUND",
+		 "message": "image not found",
+		 "detail": "image not found"
+		}]
+	}
+	`
+	RandomStr                    = `random`
+	RandomJsonStr                = `{"nope":"nope"}`
+	ErrJsonWithEmptyErrorsField  = `{"errors":[]}`
+	ErrJsonWithNoMessageField    = `{"errors":[{"code":"nope","detail":"nope"}]}`
+	ErrJsonWithEmptyMessageField = `{"errors":[{"code":"nope","message":""},{"message":""}]}`
+)
+
+func TestExtractErrResponseMessage(t *testing.T) {
+	// Test set up: create the io streams for testing purposes
+	// multiple streams needed: these streams only have read ends
+	singleErrTestStream := ioutil.NopCloser(bytes.NewReader([]byte(ErrJsonStr401)))
+	multipleErrTestStream := ioutil.NopCloser(bytes.NewReader([]byte(MultipleErrJsonStr)))
+	randomStrTestStream := ioutil.NopCloser(bytes.NewReader([]byte(RandomStr)))
+	malformedJSONTestStream := ioutil.NopCloser(bytes.NewReader([]byte(RandomJsonStr)))
+	emptyErrorsJSONTestStream := ioutil.NopCloser(bytes.NewReader([]byte(ErrJsonWithEmptyErrorsField)))
+	noMessageJSONTestStream := ioutil.NopCloser(bytes.NewReader([]byte(ErrJsonWithNoMessageField)))
+	emptyMessageJSONTestStream := ioutil.NopCloser(bytes.NewReader([]byte(ErrJsonWithEmptyMessageField)))
+
+	// Test 1: single error message extraction
+	msg, err := extractErrResponseMessage(singleErrTestStream)
+	assert.Nil(t, err, "test: (single error message) extraction should success for well-formatted error json")
+	assert.NotNil(t, msg, "test: (single error message) message extracted for well-formatted error json")
+	assert.Equal(t, "authentication required", msg,
+		"test: (single error message) extracted message: %s; expected: authentication required", msg)
+
+	// Test 2: multiple error message extraction
+	msg, err = extractErrResponseMessage(multipleErrTestStream)
+	assert.Nil(t, err, "test: (multiple error messages) extraction should success for well-formatted error json")
+	assert.NotNil(t, msg, "test: (multiple error messages) message extracted for well-formatted error json")
+	assert.Equal(t, "authentication required, image not found", msg,
+		"test: (multiple error messages) extracted message: %s; expected: authentication required, image not found", msg)
+
+	// Test 3: random string in the stream that is not a json
+	msg, err = extractErrResponseMessage(randomStrTestStream)
+	assert.Equal(t, "", msg, "test: (non-json string) no message should be extracted")
+	assert.NotNil(t, err, "test: (non-json string) extraction should fail")
+
+	// Test 4: malformed json string
+	msg, err = extractErrResponseMessage(malformedJSONTestStream)
+	assert.Equal(t, "", msg, "test: (malformed json string) no message should be extracted")
+	assert.NotNil(t, err, "test: (malformed json string) extraction should fail")
```

Comment:
After making this change (https://github.com/vmware/vic/pull/6067/files#r134591535), we can just `assert.Equal(t, err, errJsonFormat)` instead of the two asserts here.